tests/integration/project_test.py

```diff
@@ -1496,6 +1497,61 @@ def test_project_up_with_secrets(self):
         output = container.logs()
         assert output == b"This is the secret\n"
 
+    @v3_only()
+    def test_project_up_with_added_secrets(self):
+        node = create_host_file(self.client, os.path.abspath('tests/fixtures/secrets/default'))
+
+        config_input1 = {
+            'version': V3_1,
+            'services': [
+                {
+                    'name': 'web',
+                    'image': 'busybox:latest',
+                    'command': 'cat /run/secrets/special',
+                    'environment': ['constraint:node=={}'.format(node if node is not None else '')]
+                }
+
+            ],
+            'secrets': {
+                'super': {
+                    'file': os.path.abspath('tests/fixtures/secrets/default')
+                }
+            }
+        }
+        config_input2 = copy.deepcopy(config_input1)
+        # Add the secret
+        config_input2['services'][0]['secrets'] = [
+            types.ServiceSecret.parse({'source': 'super', 'target': 'special'})
+        ]
+
+        config_data1 = build_config(**config_input1)
+        config_data2 = build_config(**config_input2)
+
+        # First up with non-secret
+        project = Project.from_config(
+            client=self.client,
+            name='composetest',
+            config_data=config_data1,
+        )
+        project.up()
+        project.stop()
```

Comment:
I think you don't want to `stop` here, since your goal is to test that an update to an existing service works.

Reply:
Removed it in the latest commit, tests pass on my computer, so should pass on Jenkins.